### PR TITLE
provider/google: Google Cloud SQL Support

### PR DESCRIPTION
**Pull request by lwander**
Added:
* `google_sql_database`
* `google_sql_database_instance`

Documentation & tests are written for both.

A lot of the files in the website directory were touched to separate the resources in the google cloud provider documentation neatly. 

@sparkprime

### CODE REVIEW

**Comment by lwander** [comment]
What exactly is failing? The log is not very clear.

**Comment by sparkprime** [comment]
Looks like this:

2015/10/06 21:03:34 [DEBUG] vertex root.root: walking
--- FAIL: TestApply_parallelism1-2 (2.26s)
	apply_test.go:94: bad: 2.259610

File: website/source/docs/providers/google/r/storage_object_acl.html.markdown [hunk 1]
```diff
@@ -1,7 +1,11 @@
 ---
 layout: "google"
 page_title: "Google: google_storage_object_acl"
+<<<<<<< HEAD
```
Comment (sparkprime) [inline website/source/docs/providers/google/r/storage_object_acl.html.markdown]: oops

File: website/source/docs/providers/google/r/sql_database_instance.html.markdown [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the instance.
+
+* `region` - (Optional) The region the instance will sit in. Defaults to the
+  project region.
```
Comment (sparkprime) [inline website/source/docs/providers/google/r/sql_database_instance.html.markdown]: what's the project region?

File: website/source/docs/providers/google/r/sql_database_instance.html.markdown [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+* `name` - (Optional) A name for this whitelist entry.
+
+* `value` - (Optional) A CIDR notation IPv4 or IPv6 address that is allowed to
+  access this instance. Must be set even if other two attributes are not for
+  the whitelist to become active.
+
+The optional `settings.location_preference` subblock supports:
+
+* `follow_gae_application` - (Optional) A GAE application whose zone to remain
```
Comment (sparkprime) [inline website/source/docs/providers/google/r/sql_database_instance.html.markdown]: Spell out Google App Engine (GAE) the first time the acronym appears.

File: website/source/docs/providers/google/r/sql_database.html.markdown [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+resource "google_sql_database_instance" "master" {
+	name = "master-instance"
+	
+    settings {
+        tier = "D0"
+    }
+}
+
+resource "google_sql_database" "users" {
+	name = "image-store-bucket"
```
Comment (sparkprime) [inline website/source/docs/providers/google/r/sql_database.html.markdown]: strange name for it

File: website/source/docs/providers/google/r/sql_database_instance.html.markdown (hunk 2 above)
Comment (lwander) [inline website/source/docs/providers/google/r/sql_database_instance.html.markdown]: I meant the provider region, I'll fix that

File: website/source/docs/providers/google/r/sql_database_instance.html.markdown (hunk 3 above)
Comment (lwander) [inline website/source/docs/providers/google/r/sql_database_instance.html.markdown]: Good point!

**Comment by lwander** [comment]
I deleted my fork to git rid of that strange merge history - I think it was caused by accidentally committing to master and then pulling from upstream. I'll reopen this PR on the new fork.

**Pull request closed without merging**